File: 0975-range-sum-of-bst/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = 0 ;
    int rangeSumBST(TreeNode* root, int low, int high) {
        check(root,low,high);
        return res;
    }
    
    void check(TreeNode* root , int low , int high){
        if(root != NULL){
        if(root->val >= low and root->val <= high)
            res += root->val ;
        
        check(root->left,low,high);
        check(root->right,low,high);
        }
    }
};

File: 0975-range-sum-of-bst/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        return solve(root,low,high);
    }

    public int solve(TreeNode root, int low, int high){
        if(root == null) return 0 ;
        return  (root.val >= low && root.val <= high ? root.val : 0 ) + solve(root.left,low,high) + solve(root.right,low,high);
    }
}
